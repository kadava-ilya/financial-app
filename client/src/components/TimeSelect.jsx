import React from "react";
import {Box, FormControl, Select, MenuItem, Typography} from "@mui/material";
import {useDispatch, useSelector} from "react-redux";
import {setTimeRange} from "../redux/slices/tickersSlice";
import {selectTimeRange} from "../redux/selectors";
import {TIME_RANGES} from "../utils/constants";

export const TimeSelect = () => {

    const dispatch = useDispatch();
    const timeRange = useSelector(selectTimeRange)

    const handleChange = (event) => {
        dispatch(setTimeRange(event.target.value));
    };

    return (
        <Box sx={{display: 'flex', alignItems: 'center',}}>
            <Typography sx={{marginRight: '20px'}}>Select the time range: </Typography>
            <FormControl>
                <Select
                    id="select"
                    value={timeRange}
                    onChange={handleChange}
                    displayEmpty
                    inputProps={{ 'aria-label': 'Without label' }}
                >
                    {
                        TIME_RANGES.map(range => (
                            <MenuItem
                                key={range.id}
                                value={range.value}
                            >
                                {range.title}
                            </MenuItem>
                        ))
                    }
                </Select>
            </FormControl>
        </Box>
    )
}