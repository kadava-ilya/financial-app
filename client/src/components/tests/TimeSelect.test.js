import {TimeSelect} from "../TimeSelect";
import {fireEvent, render, screen} from "@testing-library/react";
import * as reduxHooks from "react-redux";

jest.mock('react-redux')

const mockedUseSelector = jest.spyOn(reduxHooks, 'useSelector')

const test = [
    {
        id: 1,
        title: '1s',
        value: 1000,
    },
    {
        id: 2,
        title: '5s',
        value: 5000,
    },
]

describe('TimeSelect', () => {
    it('should create TimeSelect without options', function () {
        mockedUseSelector.mockReturnValue([])

        const component = render(<TimeSelect />)

        expect(component).toMatchSnapshot()
    });

    it('should create TimeSelect with options', function () {
        mockedUseSelector.mockReturnValue(test)

        const component = render(<TimeSelect />)

        expect(component).toMatchSnapshot()
    });

    it('should select renders in TimeSelect component', () => {
        mockedUseSelector.mockReturnValue(test)

        render(<TimeSelect />)

        expect(screen.getByText(/Select the time range:/))
    });

    // it('should select works on click in TimeSelect component', () => {
    //     mockedUseSelector.mockReturnValue(test)
    //
    //     render(<TimeSelect />)
    //     expect(screen.getByText(/5s/)).toBeInTheDocument()
    //     fireEvent.click(screen.getByText(/5s/))
    //     fireEvent.click(screen.getByText(/1s/))
    //     expect(screen.getByText(/1s/)).toBeInTheDocument()
    // });
})