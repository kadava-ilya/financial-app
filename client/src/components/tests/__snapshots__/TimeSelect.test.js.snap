// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TimeSelect should create TimeSelect without options 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="MuiBox-root css-70qvj9"
      >
        <p
          class="MuiTypography-root MuiTypography-body1 css-1y36zwc-MuiTypography-root"
        >
          Select the time range: 
        </p>
        <div
          class="MuiFormControl-root css-1nrlq1o-MuiFormControl-root"
        >
          <div
            class="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-formControl  css-1yk1gt9-MuiInputBase-root-MuiOutlinedInput-root-MuiSelect-root"
          >
            <div
              aria-expanded="false"
              aria-haspopup="listbox"
              aria-label="Without label"
              aria-labelledby="select"
              class="MuiSelect-select MuiSelect-outlined MuiOutlinedInput-input MuiInputBase-input css-11u53oe-MuiSelect-select-MuiInputBase-input-MuiOutlinedInput-input"
              id="select"
              role="button"
              tabindex="0"
            >
              <span
                class="notranslate"
              >
                ​
              </span>
            </div>
            <input
              aria-hidden="true"
              class="MuiSelect-nativeInput css-yf8vq0-MuiSelect-nativeInput"
              tabindex="-1"
              value=""
            />
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiSelect-icon MuiSelect-iconOutlined css-hfutr2-MuiSvgIcon-root-MuiSelect-icon"
              data-testid="ArrowDropDownIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M7 10l5 5 5-5z"
              />
            </svg>
            <fieldset
              aria-hidden="true"
              class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
            >
              <legend
                class="css-hdw1oc"
              >
                <span
                  class="notranslate"
                >
                  ​
                </span>
              </legend>
            </fieldset>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="MuiBox-root css-70qvj9"
    >
      <p
        class="MuiTypography-root MuiTypography-body1 css-1y36zwc-MuiTypography-root"
      >
        Select the time range: 
      </p>
      <div
        class="MuiFormControl-root css-1nrlq1o-MuiFormControl-root"
      >
        <div
          class="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-formControl  css-1yk1gt9-MuiInputBase-root-MuiOutlinedInput-root-MuiSelect-root"
        >
          <div
            aria-expanded="false"
            aria-haspopup="listbox"
            aria-label="Without label"
            aria-labelledby="select"
            class="MuiSelect-select MuiSelect-outlined MuiOutlinedInput-input MuiInputBase-input css-11u53oe-MuiSelect-select-MuiInputBase-input-MuiOutlinedInput-input"
            id="select"
            role="button"
            tabindex="0"
          >
            <span
              class="notranslate"
            >
              ​
            </span>
          </div>
          <input
            aria-hidden="true"
            class="MuiSelect-nativeInput css-yf8vq0-MuiSelect-nativeInput"
            tabindex="-1"
            value=""
          />
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiSelect-icon MuiSelect-iconOutlined css-hfutr2-MuiSvgIcon-root-MuiSelect-icon"
            data-testid="ArrowDropDownIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M7 10l5 5 5-5z"
            />
          </svg>
          <fieldset
            aria-hidden="true"
            class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
          >
            <legend
              class="css-hdw1oc"
            >
              <span
                class="notranslate"
              >
                ​
              </span>
            </legend>
          </fieldset>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
